tests/unit/Service/PagePurgerTest.php

```diff
@@ -51,4 +52,49 @@ public function testPurgePage() {
 		$this->assertTrue( $service->purge( $page ) );
 	}
 
+	public function testPurgePages() {
+		$api = $this->getMockApi();
+		$api->expects( $this->once() )
+			->method( 'postRequest' )
+			->with(
+				$this->isInstanceOf( '\Mediawiki\Api\SimpleRequest' )
+			)
+			->will( $this->returnValue(
+				[
+	    		"batchcomplete" => "",
+	    		"purge" => [
+	        	[
+	          	"ns" => 0,
+	            "title" => "Foo",
+	            "purged" => ""
+	        	],
+	        	[
+	            "ns" => 0,
+	            "title" => "Bar",
+	            "purged" => ""
+	        	]
+	    		]
+	    	]
+			) );
+
+			$service = new PagePurger( $api );
+
+			$pages = new Pages( [
+				new Page(
+				new PageIdentifier(
+					new Title( 'Foo', 0 ),
+					123
+				),
+				new Revisions( [] )
+			), new Page(
+				new PageIdentifier(
+					new Title( 'Bar', 1 ),
+					123
+				),
+				new Revisions( [] )
```

Comment:
If no `$revisions` parameter is passed, a new Revisions will be created in the Page constructor — so don't worry about passing this in.